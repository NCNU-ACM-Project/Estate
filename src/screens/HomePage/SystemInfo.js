import React from "react";
import { Text, View, StyleSheet, ScrollView } from "react-native";
import NormalText from "../../components/NormalText";
import Hr from "../../components/Hr";
import SystemInfoEntry from "./SystemInfoEntry";
import { useSelector } from "react-redux";

const SystemInfo = ({ style }) => {
  const { systemInfo } = useSelector((state) => state.home);
  return (
    <ScrollView style={[styles.container, style]}>
      <NormalText style={styles.titleText}>系統公告</NormalText>
      <Hr />
      <SystemInfoEntry systemInfo={systemInfo} />
    </ScrollView>
  );
};

const styles = StyleSheet.create({
  container: {
    borderColor: "#000",
    borderWidth: 1,
    margin: 10,
    borderRadius: 20,
    paddingLeft: 10,
    paddingRight: 10,
  },
  titleText: {
    textAlign: "center",
    width: "100%",
    fontSize: 20,
  },
});

export default SystemInfo;
