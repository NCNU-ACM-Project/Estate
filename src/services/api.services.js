// create a axios instance
import axios from "axios";

export const houseRequest = axios.create({
    baseURL: "http://192.168.0.120:3001/",
    timeout: 1000,
    headers: { "content-type": "application/json" },
  });


// this is example
// export const create_student_with_fetch = async (name) => {
// return fetch("http://192.168.0.120:3001/students", {
//     body: `{"name": "${name}"}`,
//     headers: { "content-type": "application/json" },
//     method: "POST",
// })
//     .then((res) => res.json())
//     .then((data) => {
//     console.log(data);
//     return data;
//     });
// };

// this is also a example
// export const read_student_with_axios = async () => {
//     return userRequest
//       .get("/students")
//       .then((res) => res.data)
//       .then((data) => {
//         // console.log(data);
//         return data;
//       });
//   };