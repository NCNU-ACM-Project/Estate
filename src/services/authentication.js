import axios from "axios";

const backendUrl = `${process.env.REACT_APP_API_URL}`;

export async function login (username, password) {
    // may check if local token expire and local login status at next version or not

    // use api to login
    return await apiAuthenticate(username, password);
}

export function logout () {
    // may revoke the token store in phone
}

async function apiAuthenticate(username, password) {
    return accountDetail = await axios.post(`${backendUrl}/account`, {username, password});
}