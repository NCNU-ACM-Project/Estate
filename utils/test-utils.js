import * as React from 'react'
import { render } from '@testing-library/react-native'
import { NativeBaseProvider } from 'native-base'
import { Provider } from 'react-redux';
import { store } from '../src/store';

// not sure how, but use 2 more promose will cause error, just replace the global Promise
// ref https://github.com/callstack/react-native-testing-library/issues/379#issuecomment-762597260
import Promise from "promise-polyfill";
global.Promise = Promise;

const AllTheProviders = ({children}) => {
  const inset = {
    frame: { x: 0, y: 0, width: 0, height: 0 },
    insets: { top: 0, left: 0, right: 0, bottom: 0 },
  };

  return ( 
    <Provider store={store}>
      <NativeBaseProvider initialWindowMetrics={inset}>
        {children}
      </NativeBaseProvider>
    </Provider>
  )
}

const customRender = (ui, options) =>
  render(ui, {wrapper: AllTheProviders, ...options})

// re-export everything
export * from '@testing-library/react-native'

// override render method
export {customRender as render}