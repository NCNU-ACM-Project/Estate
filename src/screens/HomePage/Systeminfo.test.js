import React from "react";
import { render, fireEvent, cleanup, waitFor } from "test-utils";
import MockAdapter from "axios-mock-adapter";
import { houseRequest } from "../../../src/services/api.services";
import SystemInfo from "./SystemInfo";

// const mock = new MockAdapter(userRequest, { delayResponse: 500 });

describe("HomePage", () => {
  it("systeminfo's snapshot renders correctly", () => {
    const tree = render(<SystemInfo />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe("test functionality", () => {
  //   beforeEach(() => {
  //     mock.onGet("http://192.168.0.120:3001/students").reply(200, [{ id: 1, name: "John Smith" }]);
  //   });

  //   afterEach(() => {
  //     mock.reset();
  //     cleanup();
  //   });

//   test("test whether the data fetch when click read from axios", async () => {
//     const { getByText, findByText, toJSON } = render(<DemoPage />);

//     const axiosGetBtn = getByText("read(axios)");
//     fireEvent(axiosGetBtn, "press");
//     await findByText('"loading"');
//     await findByText('"John Smith"');
//   });
});
