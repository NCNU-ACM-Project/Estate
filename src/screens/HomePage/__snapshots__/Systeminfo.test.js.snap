// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage systeminfo's snapshot renders correctly 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <RCTScrollView
    style={
      Array [
        Object {
          "borderColor": "#000",
          "borderRadius": 20,
          "borderWidth": 1,
          "margin": 10,
          "paddingLeft": 10,
          "paddingRight": 10,
        },
        undefined,
      ]
    }
  >
    <View>
      <Text
        style={
          Array [
            Object {
              "fontSize": 24,
            },
            Object {
              "fontSize": 50,
            },
            Object {
              "fontSize": 20,
              "textAlign": "center",
              "width": "100%",
            },
          ]
        }
      >
        系統公告
      </Text>
      <View
        style={
          Array [
            Object {
              "backgroundColor": "black",
              "height": 1,
              "margin": 3,
            },
            undefined,
          ]
        }
      />
      <View
        style={
          Object {
            "flex": 1,
            "flexDirection": "column",
            "justifyContent": "flex-start",
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "flex-start",
              "flex": 1,
              "flexDirection": "row",
            }
          }
        >
          <Text
            style={
              Array [
                undefined,
                undefined,
                Object {
                  "flex": 2,
                },
              ]
            }
          >
            7 月8 日2021 年
          </Text>
          <View
            style={
              Object {
                "alignSelf": "stretch",
                "backgroundColor": "#000",
                "display": "flex",
                "margin": 3,
                "width": 1,
              }
            }
          />
          <Text
            style={
              Array [
                undefined,
                undefined,
                Object {
                  "flex": 9,
                },
              ]
            }
          >
            werawerkwejrwalejrklwejrlwerjlwekfjelwjklfawefawjefklawjefklajflwafejwkl
          </Text>
        </View>
        <View
          style={
            Object {
              "alignItems": "flex-start",
              "flex": 1,
              "flexDirection": "row",
            }
          }
        >
          <Text
            style={
              Array [
                undefined,
                undefined,
                Object {
                  "flex": 2,
                },
              ]
            }
          >
            7 月9 日2021 年
          </Text>
          <View
            style={
              Object {
                "alignSelf": "stretch",
                "backgroundColor": "#000",
                "display": "flex",
                "margin": 3,
                "width": 1,
              }
            }
          />
          <Text
            style={
              Array [
                undefined,
                undefined,
                Object {
                  "flex": 9,
                },
              ]
            }
          >
            werawerkwejrwalejrklwejrlwerjlwekfjelwjklfawefawjefklawjefklajflwafejwkl
          </Text>
        </View>
        <View
          style={
            Object {
              "alignItems": "flex-start",
              "flex": 1,
              "flexDirection": "row",
            }
          }
        >
          <Text
            style={
              Array [
                undefined,
                undefined,
                Object {
                  "flex": 2,
                },
              ]
            }
          >
            7 月10 日2021 年
          </Text>
          <View
            style={
              Object {
                "alignSelf": "stretch",
                "backgroundColor": "#000",
                "display": "flex",
                "margin": 3,
                "width": 1,
              }
            }
          />
          <Text
            style={
              Array [
                undefined,
                undefined,
                Object {
                  "flex": 9,
                },
              ]
            }
          >
            werawerkwejrwalejrklwejrlwerjlwekfjelwjklfawefawjefklawjefklajflwafejwkl
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>
</RNCSafeAreaProvider>
`;
